src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -176,11 +193,15 @@ private void removeDeadDataNodes() {
     return getHashMap(DATANODES_KEY);
   }
 
-  public Collection<String> getAllTaskIds() {
-    HashMap<String, String> allTasksIds = getJournalNodes();
-    allTasksIds.putAll(getNameNodes());
-    allTasksIds.putAll(getDataNodes());
-    return allTasksIds.values();
+  public Set<String> getAllTaskIds() {
+    Set<String> allTaskIds = new HashSet<String>();
+    Collection<String> journalNodes = getJournalNodes().values();
+    Collection<String> nameNodes = getNameNodes().values();
+    Collection<String> dataNodes = getDataNodes().values();
+    allTaskIds.addAll(journalNodes);
+    allTaskIds.addAll(nameNodes);
```

Comment:
They are separate tasks and we need to track them as well. This was not implemented initially with the HA features. I guess the thought was that most of the time if the ZKFC's go down, they will throw exceptions and bring down the NN's. Still, this case needs full coverage. I added a TODO for this.